src/shogun/evaluation/SigmoidCalibration.h

```diff
@@ -0,0 +1,112 @@
+ * those
+ * of the authors and should not be interpreted as representing official
+ * policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#ifndef _SIGMOID_CALIBRATION_H__
+#define _SIGMOID_CALIBRATION_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/evaluation/Calibration.h>
+#include <shogun/mathematics/Statistics.h>
+
+
+namespace shogun
+{
+	/** @brief Calibrates labels based on Platt Scaling [1]. Note that first calibration parameters need to be fitted by calling fit. 
+	* Usually this done using the training data and labels. 
+	* [1] Platt J. Probabilistic outputs for support vector machines and comparisons to regularized likelihood methods.
+	* Advances in large margin classifiers. 1999
+	*/
+	class CSigmoidCalibration : public CCalibration
+	{
+	public:
+		/** Constructor. */
+		CSigmoidCalibration();
+
+		/** Destructor. */
+		virtual ~CSigmoidCalibration();
+
+		/** Get name. */
+		virtual const char* get_name() const
+		{
+			return "SigmoidCalibration";
+		}
+
+		/** Fit sigmoid parameters for binary labels.
+		* @param predictions The predictions outputted by the machine
+		* @param targets The true labels corresponding to the predictions
+		* @return boolean indicating whether the calibration was succesful
+		**/
+		virtual bool fit_binary(CBinaryLabels* predictions, CBinaryLabels* targets);
+
+		/** Calibrate binary predictions based on parameters learned by calling fit.
+		* @param predictions The predictions outputted by the machine
+		* @return Calibrated binary labels
+		**/
+		virtual CBinaryLabels* calibrate_binary(CBinaryLabels* predictions);
+
+		/** Fit calibration parameters for multiclass labels. Fits sigmoid 
+		* parameters for each class seperately.
+		* @param predictions The predictions outputted by the machine
+		* @param targets The true labels corresponding to the predictions
+		* @return boolean indicating whether the calibration was succesful
+		**/
+		virtual bool fit_multiclass(CMulticlassLabels* predictions, CMulticlassLabels* targets);
+
+		/** Calibrate multiclass predictions based on parameters learned by calling fit.
+		* The predictions are normalized over all classes.
+		* @param predictions The predictions outputted by the machine
+		* @return Calibrated binary labels
+		**/
+		virtual CMulticlassLabels* calibrate_multiclass(CMulticlassLabels* predictions);
+
+	private:
+		/** Initialize parameters */
+		void init();
+
+		/** Helper function that calibrates values of given vector using the given sigmoid parameters
+		* @param values The values to be calibrated
+		* @param params The sigmoid paramters to be used for calibration
+		*/
+		SGVector<float64_t> calibrate_values(SGVector<float64_t> values, CStatistics::SigmoidParamters params);
+
+	private:
+		/** Array to store sigmoid parameters for each class. In case of binary labels, only one pair of parameters are stored. */
+		CStatistics::SigmoidParamters* m_sigmoid_parameters;
```

Comment:
wouldn't storing this in an std::vector would be better?

Reply:
or in worst case DynamicObjectArray, for assuring ref counting

Reply:
I will replace this with `std::vector`... DynamicObjectArray seems to support only CSGObjects, it can't hold structs like `CStatistics::SigmoidParamters`. Please correct me if I'm wrong about that :)